tasks/src/main/java/error_exceptions/service/StudentsContainer.java

```diff
@@ -0,0 +1,134 @@
+    public static final Student studentMillsVincent = new Student("Mills", "Vincent");
+    public static final Student studentAndrewsBetty = new Student("Andrews", "Betty");
+    public static final Student studentHarrisJulian = new Student("Harris", "Julian");
+    public static final Student studentBensonMary = new Student("Benson", "Mary");
+    public static final Student studentWadeAnthony = new Student("Wade", "Anthony");
+
+    private static double averageMark = 0.0;
+    private static int countSubjects = 0;
+    private static double sumAllMarks = 0.0;
+
+    private static double averageMark(Student student) {
+        sumAllMarks += student.getMark();
+        if (student.getSubject() != null) {
+            countSubjects++;
+        }
+        return sumAllMarks / countSubjects;
+    }
+
+    public static double averageMarkOfAllSubjects(List<Student> students, String lastName,
+        String firstName) {
+        for (Student student : students) {
+            if (student.getLastName().equals(lastName) && student.getFirstName()
+                .equals(firstName)) {
+                averageMark = averageMark(student);
+            }
+        }
+        if (averageMark == 0.0) {
+            throw new StudentSubjectsException("Student " + lastName + " " + firstName
+                + " without marks or not study in this university");
+        }
+        return averageMark;
+    }
+
+    private static boolean studentInGroup(List<Faculty> faculties, Group.GroupName group) {
+        boolean hasStudent = false;
+        for (Faculty faculty : faculties) {
+            String groupNames = faculty.getGroupName().name();
+            if (groupNames.contains(group.name())) {
+                hasStudent = true;
+                break;
+            }
+        }
+        return hasStudent;
+    }
+
+    private static boolean groupsInFaculty(List<Faculty> faculties, Group.GroupName group,
+        Faculty.FacultyName facultyName) {
+        boolean hasGroup = false;
+        for (Faculty faculty : faculties) {
+            if (faculty.getFacultyName().equals(facultyName) && faculty.getGroupName()
+                .equals(group)) {
+                hasGroup = true;
+                break;
+            }
+        }
+        return hasGroup;
+    }
+
+    private static boolean facultyInUniversity(List<Faculty> faculties,
+        Faculty.FacultyName facultyName) {
+        boolean hasFaculty = false;
+        for (Faculty faculty : faculties) {
+            String facultyNames = faculty.getFacultyName().name();
+            if (facultyNames.contains(facultyName.name())) {
+                hasFaculty = true;
+                break;
+            }
+        }
+        return hasFaculty;
+    }
+
+    public static double averageMarkOfSpecificSubjectGroupFaculty(List<Faculty> faculties,
+        List<Student> students, Student.SubjectsName subject,
+        Group.GroupName group, Faculty.FacultyName facultyName) {
+        sumAllMarks = 0;
```

Comment:
That is not a good idea to reset your class fields inside this method. It can affect other methods that use the fields like averageMark